File: languages/php/Bloock/Entity/Key/Key.php
<?php

namespace Bloock\Entity\Key;

use Exception;
use Bloock\Entity\Key\LocalKey;
use Bloock\Entity\Key\ManagedKey;

/**
 * Represents a key entity that can be either a ManagedKey or a LocalKey.
 */
class Key
{
    public ?LocalKey $localKey = null;
    public ?ManagedKey $managedKey = null;

    /**
     * Creates a Key instance with a local or managed key.
     * @param $key
     * @throws Exception
     */
    public function __construct($key)
    {
        if ($key instanceof LocalKey) {
            $this->localKey = $key;
        } else if ($key instanceof ManagedKey) {
            $this->managedKey = $key;
        } else {
            throw new Exception("Invalid key provided");
        }
    }

    public function toProto(): \Bloock\Key
    {
        $s = new \Bloock\Key();

        if ($this->localKey != null) {
            $s->setLocalKey($this->localKey->toProto());
        }

        if ($this->managedKey != null) {
            $s->setManagedKey($this->managedKey->toProto());
        }

        return $s;
    }
}

File: languages/python/bloock/entity/key/key.py
import bloock._bridge.proto.keys_entities_pb2 as proto
from bloock.entity.key.local_key import LocalKey
from bloock.entity.key.managed_key import ManagedKey


class Key:
    """
    Represents a key entity that can be either a ManagedKey or a LocalKey.
    """
    local_key = None
    managed_key = None

    def __init__(self, key) -> None:
        """
        Constructs a Key object for a given managed or local key object.
        :type key: object
        :rtype: object
        """
        if isinstance(key, LocalKey):
            self.local_key = key
        elif isinstance(key, ManagedKey):
            self.managed_key = key
        else:
            raise Exception(
                "Invalid key provided. Must be of type LocalKey or ManagedKey")

    def to_proto(self) -> proto.Key:
        local_key = None
        if self.local_key is not None:
            local_key = self.local_key.to_proto()

        managed_key = None
        if self.managed_key is not None:
            managed_key = self.managed_key.to_proto()

        return proto.Key(
            local_key=local_key,
            managed_key=managed_key,
        )
